core/lib/Cardano/Address/Script.hs

```diff
@@ -504,14 +509,25 @@ instance FromJSON Cosigner where
     parseJSON = withObject "Cosigner" $ \o ->
         Cosigner <$> o .: "cosigner"
 
+instance ToJSON XPub where
+    toJSON = String . T.decodeUtf8 . encode EBase16 . xpubToBytes
+
+instance FromJSON XPub where
+    parseJSON = withText "XPub" $ \txt ->
+        case fromBase16 (T.encodeUtf8 txt) of
+            Left err -> fail err
+            Right hex -> case xpubFromBytes hex of
+                Nothing -> fail "Extended public key cannot be retrieved from a given hex bytestring"
+                Just validXPub -> pure validXPub
```

Comment:
I'd suggest making those simple functions :`encodeXPub :: XPub -> Aeson.Value` and `parseXPub :: Aeson.Value -> Aeson.Parser XPub` instead to avoid the orphan instance.

Reply:
done, see state as in https://github.com/input-output-hk/cardano-addresses/pull/99/commits/070e6bf1e09840eb7f88e782baed7eeee56dcabe